api/src/pcapi/core/offers/api.py

```diff
@@ -193,6 +193,10 @@ def _get_internal_accessibility_compliance(venue: offerers_models.Venue) -> dict
 
 def create_draft_offer(body: schemas.PostDraftOfferBodyModel, venue: offerers_models.Venue) -> models.Offer:
     validation.check_offer_subcategory_is_valid(body.subcategory_id)
+    ean_code = None
+    if body.extra_data is not None:
+        ean_code = body.extra_data.get("ean", None)
+    validation.check_product_ean_valide_for_record_store_venue(body.subcategory_id, ean_code, venue.venueTypeCode)
```

Comment:
Aujourd'hui c'est pour les disquaires mais potentiellement demain on fera d'autres vérifs (sur les livres, instruments de musiques etc.) du coup je dirais qu'on peut l'appeler

```suggestion
    validation.check_product_ean_for_venue_and_subcategory(body.subcategory_id, ean_code, venue.venueTypeCode)
```

(voir juste check_offer_product_ean )